### ISSUE TITLE
Support multi-agent

### ISSUE THREAD

**Issue by lzcrwxl** [langgenius/dify#11882]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

I would like to use multi-agent functionality in a custom scenario I create.Is there any plan to support multi-agent functionality?

### 3. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

**Comment by Taarjax** [comment]
Hey, I'd also like to know if there's a way to create a multi-agent system with a main agent acting as an orchestrator and several sub-agents, each equipped with their own tools, all maintaining some form of contextual memory. 

Perhaps this could be achieved by establishing a workflow where the main agent makes HTTP requests to the sub-agents? I'm open to other suggestions as well. 

If anyone has experience with this, I would appreciate your feedback. 
Thanks a lot.

**Comment by huynhbaobk** [comment]
Same here, really looking forward to this feature!

**Comment by Taarjax** [comment]
> Hi, [@lzcrwxl](https://github.com/lzcrwxl). I'm [Dosu](https://dosu.dev), and I'm helping the Dify team manage their backlog. I'm marking this issue as stale.
> 
> **Issue Summary**
> 
> * You requested support for multi-agent functionality in a custom scenario and expressed interest in contributing.
> * [@Taarjax](https://github.com/Taarjax) suggested a workflow involving HTTP requests for a multi-agent system and noted the roadmap includes an 'Agent' node in Workflow.
> * The suggestion received positive feedback from other users, including [@huynhbaobk](https://github.com/huynhbaobk).
> 
> **Next Steps**
> 
> * Please let us know if this issue is still relevant to the latest version of the Dify repository by commenting here to keep the discussion open.
> * If there is no further activity, this issue will be automatically closed in 15 days.
> 
> Thank you for your understanding and contribution!

Yes, this is still relevant today, but it's still complicated, if not impossible, to use multi-agent natively in Dify.

**Comment by saendu** [comment]
Will this also be available for the self-hosted version? I am a bit confused with agent strategies and the dependency to the marketplace.
